packages/mui-base/src/Select/Root/SelectRoot.test.tsx

```diff
@@ -0,0 +1,242 @@
+import * as React from 'react';
+import * as Select from '@base_ui/react/Select';
+import { fireEvent, flushMicrotasks, screen } from '@mui/internal-test-utils';
+import { createRenderer } from '#test-utils';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import userEvent from '@testing-library/user-event';
+
+const user = userEvent.setup();
+
+describe('<Select.Root />', () => {
+  const { render } = createRenderer();
+
+  describe('prop: defaultValue', () => {
+    it('should select the option by default', async () => {
+      await render(
+        <Select.Root defaultValue="b" animated={false}>
+          <Select.Trigger data-testid="trigger">
+            <Select.Value />
+          </Select.Trigger>
+          <Select.Positioner>
+            <Select.Popup>
+              <Select.Option value="a">a</Select.Option>
+              <Select.Option value="b">b</Select.Option>
+            </Select.Popup>
+          </Select.Positioner>
+        </Select.Root>,
+      );
+
+      const trigger = screen.getByTestId('trigger');
+
+      fireEvent.click(trigger);
+
+      await flushMicrotasks();
+
+      expect(screen.getByRole('option', { name: 'b', hidden: false })).to.have.attribute(
+        'data-selected',
+        'true',
+      );
+    });
+  });
+
+  describe('prop: value', () => {
+    it('should select the option specified by the value prop', async () => {
+      await render(
+        <Select.Root value="b" animated={false}>
+          <Select.Trigger data-testid="trigger">
+            <Select.Value />
+          </Select.Trigger>
+          <Select.Positioner>
+            <Select.Popup>
+              <Select.Option value="a">a</Select.Option>
+              <Select.Option value="b">b</Select.Option>
+            </Select.Popup>
+          </Select.Positioner>
+        </Select.Root>,
+      );
+
+      const trigger = screen.getByTestId('trigger');
+
+      fireEvent.click(trigger);
+
+      await flushMicrotasks();
+
+      expect(screen.getByRole('option', { name: 'b', hidden: false })).to.have.attribute(
+        'data-selected',
+        'true',
+      );
+    });
+
+    it('should update the selected option when the value prop changes', async () => {
+      const { rerender } = await render(
+        <Select.Root value="a" animated={false}>
+          <Select.Trigger data-testid="trigger">
+            <Select.Value />
+          </Select.Trigger>
+          <Select.Positioner>
+            <Select.Popup>
+              <Select.Option value="a">a</Select.Option>
+              <Select.Option value="b">b</Select.Option>
+            </Select.Popup>
+          </Select.Positioner>
+        </Select.Root>,
+      );
+
+      const trigger = screen.getByTestId('trigger');
+
+      fireEvent.click(trigger);
+
+      await flushMicrotasks();
+
+      expect(screen.getByRole('option', { name: 'a', hidden: false })).to.have.attribute(
+        'data-selected',
+        'true',
+      );
+
+      rerender(
```

Comment:
Why not call `setProps()` so we don't have to duplicate the rest of the component code?